server/image_remove.go

```diff
@@ -42,16 +45,11 @@ func (s *Server) RemoveImage(ctx context.Context, req *pb.RemoveImageRequest) (r
 	for _, img := range images {
 		err = s.StorageImageServer().UntagImage(s.systemContext, img)
 		if err != nil {
-			logrus.Debugf("error deleting image %s: %v", img, err)
+			logrus.Infof("%s: error deleting image %s: %v", operation, img, err)
 			continue
 		}
-		deleted = true
 		break
 	}
-	if !deleted && err != nil {
-		return nil, err
```

Comment:
I'm still a bit scared about removing the errors entirely. In case an image has already been removed (or doesn't exist in general), we should get a `containers/image/storage.ErrNoSuchImage` (https://github.com/containers/image/blob/master/storage/storage_reference.go#L56) and only those should be ignored while others should be returned.

@nalind @mtrmac PTAL

Reply:
The issue originates from the flaking critest suite when running in parallel. The error I saw there was mainly "layer not known" because another test already removed the layer:
https://github.com/containers/storage/blob/4442d3aa2f8218e930c389535d26496edd0e07e6/errors.go#L15

Reply:
This sounds like a bug in c/storage. @nalind, @rhatdan, @giuseppe can you have a look at that? I'll be heading to the train station in a bit.

Reply:
I agree that a “layer not known” seems like a problem in c/storage if the caller is not manipulating layers individually.

> I'm still a bit scared about removing the errors entirely. In case an image has already been removed (or doesn't exist in general), we should get a containers/image/storage.ErrNoSuchImage (https://github.com/containers/image/blob/master/storage/storage_reference.go#L56) and only those should be ignored while others should be returned.

Yes, ideally. OTOH, with all the abstraction layers involved and all the helpful flexible helpers, it’s not _trivially_ obvious that the destination is always c/storage. AFAICS it _is_ the case, but only because `ResolveNames` enforces the input to be an image ID or a docker reference; without that, `UntagImage` would happily try to remove a `dir:latest`.

Reply:
Ewww. Well, we still cannot just ignore the errors here.